tests/nested/manual/optee-fde/task.yaml

```diff
@@ -0,0 +1,76 @@
+summary: Test snapd OPTEE integration
+
+details: |
+    Test that snapd will use a well known OPTEE TA (trusted application) as the
+    sealing method for FDE. The snaps test-qemu-optee-kernel and
+    test-qemu-optee-gadget are custom snaps that help exercise this
+    functionality.
+
+systems: [ubuntu-24.04-arm-64]
+
+environment:
+    NESTED_CUSTOM_MODEL: $TESTSLIB/assertions/nested-24-arm64-optee.model
+    NESTED_BUILD_SNAPD_FROM_CURRENT: true
+    NESTED_REPACK_KERNEL_SNAP: false
+    NESTED_REPACK_GADGET_SNAP: false
+    NESTED_REPACK_BASE_SNAP: false
+    NESTED_SECURE_ARM_MACHINE: true
+    NESTED_CPUS: 4
+    NESTED_MEM: 4096
+
+prepare: |
+    tests.pkgs install u-boot-tools
+
+    # specific revisions are used to help allow development on the snaps without
+    # breaking current/old tests.
+    snap download --revision=2 --basename test-qemu-optee-kernel test-qemu-optee-kernel
+    snap download --revision=4 --basename test-qemu-optee-gadget test-qemu-optee-gadget
```

Comment:
Maybe point out that these snaps are created by using the scripts in https://github.com/canonical/optee-uc-fde/